File: models/IADBLocationElevation.m
//
//  IADBLocationElevation.m
//  airportsdb
//
//  Created by Christopher Hobbs on 6/17/14.
//  Copyright (c) 2014 Toonsy Net. All rights reserved.
//

#import "IADBLocationElevation.h"
#import "IADBConstants.h"

@implementation IADBLocationElevation

@dynamic name;
@dynamic type;
@dynamic elevationFeet;

//meters
-(CLLocationDistance) elevationForLocation {
    //[self respondsToSelector:@selector(elevationFeet)]
    return self.elevationFeet ? [self.elevationFeet doubleValue]/FEET_PER_METER : [super elevationForLocation];
}

@end

File: models/IADBLocationElevation.swift
//
//  IADBLocationElevation.swift
//  iAirportsDB
//
//  Created by Christopher Hobbs on 8/23/16.
//  Copyright © 2016 Toonsy Net. All rights reserved.
//

import Foundation
import CoreLocation

class IADBLocationElevation: IADBLocation {
    @NSManaged var name: String
    @NSManaged var type: String
    @NSManaged var elevationFeet: NSNumber?
    
    //meters
    override func elevationForLocation() -> CLLocationDistance {
        if let feet = self.elevationFeet {
            return feet.doubleValue / IADBConstants.feetPerMeter
        }
        return super.elevationForLocation()
    }
}
